subprojects/docs/src/docs/userguide/groovy_plugin.adoc

```diff
@@ -253,6 +253,39 @@ include::sample[dir="groovy/compilationAvoidance/groovy", files="build.gradle[ta
 include::sample[dir="groovy/compilationAvoidance/kotlin", files="build.gradle.kts[tags=groovy-compilation-avoidance]"]
 ====
 
+[[sec:incremental_groovy_compile]]
+== Incremental Groovy compilation
+
+Since 5.6, Gradle introduces an experimental incremental Groovy compiler. To enable incremental compilation for Groovy, you need:
+
+* Enable <<groovy_plugin.adoc#sec:groovy_compilation_avoidance,Groovy compilation avoidance>>.
+* Explicitly enable incremental Groovy compilation in the build script:
+
+.Enable Groovy compilation
+====
+include::sample[dir="groovy/incrementalCompilation/groovy",files="build.gradle[tags=enable-groovy-incremental]"]
+include::sample[dir="groovy/incrementalCompilation/kotlin",files="build.gradle.kts[tags=enable-groovy-incremental]"]
+====
+
+This gives you the following benefits:
+
+* Incremental builds are much faster.
+* If only a small sets of Groovy source files are changed, only the affected source files will be recompiled.
+  Classes that don't need to be recompiled remain unchanged in the output directory.
+  For example, if you only change a few Groovy test classes, you don't need to recompile all Groovy test source files - only the changed ones need to be recompiled.
+
+To understand how incremental compilation works, see <<java_plugin.adoc#sec:incremental_compile,Incremental Java compilation>> for a detailed overview.
+Note that there're several differences from Java incremental compilation:
+
+* Unlike Java, Groovy compiler doesn't inline constants, thus changes to constants won't trigger a full recompilation.
+* Groovy compiler doesn't keep `@Retention` in generated bytecode, thus changes to annotations won't trigger a full recompilation.
+
+[[sec:incremental_groovy_compilation_known_issues]]
+=== Known issues
+
+Also see <<java_plugin.adoc#sec:incremental_compilation_known_issues,Known issues for incremental Java compilation>>.
+
+* Changes to resources won't trigger a recompilation, this might result in some incorrectness - for example https://mrhaki.blogspot.com/2013/01/groovy-goodness-adding-extra-methods.html[Extension Modules].
```

Comment:
I think this a problem for compilation avoidance, not for incremental compilation, right? Or is it for both?

Reply:
I guess for both...